ehitchhiking-ui/src/app/shared/components/trips-modal/trips-modal.component.ts

```diff
@@ -12,23 +13,41 @@ import { NUMBER_OF_TRIPS_VISIBLE_ON_PAGE} from "@shared/constants/modal-constant
 	providers: [TripsModalService],
 })
 export class TripsModalComponent implements OnInit {
-  limit = NUMBER_OF_TRIPS_VISIBLE_ON_PAGE ;
+	limit = NUMBER_OF_TRIPS_VISIBLE_ON_PAGE;
 	tripsArray = [];
 	tripsArrayLenght = 0;
 	loaderSize: LoaderSize = LoaderSize.Large;
 	loading = true;
 	scrollObserver: IntersectionObserver;
-  order = 0;
-  role = {roleField: 'role', isEnable: false};
-  selectedRole: number;
-  selectedFavorite = false;
-  selectedSortByRating = false;
-  selectedBySort = SortState.None;
-
-  roles = [{value: 0, viewValue: 'Passenger'}, {value: 1, viewValue: 'Driver'}, {value: 2, viewValue: 'All'}];
+	order = 0;
+	role = {roleField: 'role', isEnable: false};
+	selectedRole: number;
+	selectedFavorite = false;
+	selectedSortByRating = false;
+	selectedBySort = SortState.None;
+	selectedByStatus = [];
+	selectedByRating = [];
 
+	roles = [{value: 0, viewValue: 'Passenger'}, {value: 1, viewValue: 'Driver'}, {value: 2, viewValue: 'All'}];
+	statuses = new FormControl();
+	ratings = new FormControl();
+	statusesTrip = [
+		{value: 0, viewValue: 'Active'},
+		{value: 1, viewValue: 'Completed'},
+		{value: 2, viewValue: 'Declined'},
+	];
+	ratesTrip = [
+		{value: 5, viewValue: '5'},
+		{value: 4, viewValue: '4'},
+		{value: 3, viewValue: '3'},
+		{value: 2, viewValue: '2'},
+		{value: 1, viewValue: '1'},
+		{value: 0, viewValue: 'not rated'},
+	];
+	statusFilterConfig = {fieldName: 'status', isEnabled: false};
+	ratingFilterConfig = {fieldName: 'rating', isEnabled: false};
```

Comment:
pls get ratingFilterConfig, statusFilterConfig, ratesTrip, statusesTrip, roles, role  form service

Reply:
in trips-modal.service.ts